### PR TITLE
config: Document settings in wc-config.yaml

### PR DESCRIPTION
**Pull request by OlleLarsson**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
Part of
- https://github.com/elastisys/compliantkubernetes/issues/8
- https://github.com/elastisys/ck8s-apps/issues/228

**Public facing documentation PR** *(if applicable)*

**Special notes for reviewer**:
I've used [kube-prometheus-stack](https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/values.yaml) as reference.

**Checklist:**

- [ ] Added relevant notes to [WIP-CHANGELOG.md](https://github.com/elastisys/compliantkubernetes-apps/blob/main/WIP-CHANGELOG.md)
- [ ] Proper commit message prefix on all commits
- [ ] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
    - [ ] is completely transparent, will not impact the workload in any way.
    - [ ] requires running a migration script.
    - [ ] will create noticeable cluster degradation.
          E.g. logs or metrics are not being collected or Kubernetes API server
          will not be responding while upgrading.
    - [ ] requires draining and/or replacing nodes.
    - [ ] will change any APIs.
          E.g. removes or changes any CK8S config options or Kubernetes APIs.
    - [ ] will break the cluster.
          I.e. full cluster migration is required.

### CODE REVIEW

File: config/config/wc-config.yaml [hunk 1]
```diff
@@ -4,99 +4,187 @@ global:
   clusterName: ${CK8S_ENVIRONMENT_NAME}-wc
   baseDomain: "set-me"
   opsDomain: "set-me"
+
+  ## Default cert-manager issuer to use in ingresses
   issuer: letsencrypt-staging
+
+  ## Verify ingress certificates
+  ##
```
Comment (viktor-f) [inline config/config/wc-config.yaml:12]: I think you can drop this and all other lines with only `##`. It will at least make the file a bit shorter.
Unless you have some reason for having them there.
```suggestion
  ## Verify ingress certificates
```

File: config/config/wc-config.yaml [hunk 2]
```diff
@@ -156,80 +302,152 @@ ck8sdash:
...
+
+    ## Node labels for pod assignment
+    ## ref: https://kubernetes.io/docs/user-guide/node-selection/
+    ##
     nodeSelector: {}
+
     config:
       useProxyProtocol: "set-me"
+
+    ## Use host ports 80 and 443
```
Comment (viktor-f) [inline config/config/wc-config.yaml:376]:
```suggestion
    ## Use host ports 80 or 443
```
You can't use both right?

Comment (OlleLarsson) [reply config/config/wc-config.yaml:376]: https://github.com/kubernetes/ingress-nginx/blob/master/charts/ingress-nginx/values.yaml#L56-L63

File: config/config/wc-config.yaml [hunk 3]
```diff
@@ -4,99 +4,187 @@ global:
   clusterName: ${CK8S_ENVIRONMENT_NAME}-wc
   baseDomain: "set-me"
   opsDomain: "set-me"
```
Comment (viktor-f) [inline config/config/wc-config.yaml:6]: The global variables could also get some simple comments

File: config/config/wc-config.yaml [hunk 4]
```diff
@@ -105,41 +193,96 @@ elasticsearch:
...
+  ## Affinity for pod assignment
+  ## ref: https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity
+  ##
   affinity: {}
+
+  ## Extra fluentd config to mount
+  ##
   extraConfigMaps: {}
+
   user:
```
Comment (lentzi90) [inline config/config/wc-config.yaml:233]: This one I think it would be good to comment on. Something like
> Configuration for the user-controllable fluentd that ships logs from nodes that the user can run pods on.

File: config/config/wc-config.yaml [hunk 5]
```diff
@@ -105,41 +193,96 @@ elasticsearch:
   clientNode:
     count: 1
 
+## Configuration for fluentd
+##
```
Comment (lentzi90) [inline config/config/wc-config.yaml:197]: It would be nice to mention here a bit more about it I think. For example that this is the "system" fluentd that runs on control-plane nodes and other "reserved" nodes that the user cannot run pods on.

File: config/config/wc-config.yaml [hunk 6]
```diff
@@ -4,99 +4,187 @@ global:
...
 
+## Open policy agent configuration
+##
 opa:
   enabled: true
+
+  ## Enable rule that requires pods to come from
+  ## the image registry defined by "URL".
+  ## "enforcement" can assume either "dryrun" or "deny".
+  ##
```
Comment (lentzi90) [inline config/config/wc-config.yaml:161]: Very good!

File: config/config/wc-config.yaml [hunk 7]
```diff
@@ -4,99 +4,187 @@ global:
...
     hostPort: "http://kube-prometheus-stack-alertmanager.monitoring:9093"
-    # if type=slack falco.alerts.slackWebhook must be set in the secrets yaml file
+    ## if type=slack falco.alerts.slackWebhook must be set in the secrets yaml file
 
+## Prometheus configuration
+##
 prometheus:
   remoteWrite:
-    # Same as 'influxDB.users.wcWriter'.
+    ## Same as 'influxDB.users.wcWriter'.
```
Comment (lentzi90) [inline config/config/wc-config.yaml:110]: This comment is confusing to me. Does it mean that it should be the same as specified for influxdb in sc-config.yaml?
I think it would make more sense to explain what it is: the user used when connecting to the remote write endpoint, by default influxdb in SC.

**Review by lentzi90 — COMMENTED** [review]
Nice start!

I think we could put a description of "standard" options at the top of the file to avoid repeating them all the time, (e.g. resources, tolerations, enabled). Other options could use some more descriptions instead. I added some comments below on some that I think could have more.

File: config/config/wc-config.yaml [hunk 8]
```diff
@@ -4,7 +4,12 @@ global:
...
   baseDomain: "set-me"
   opsDomain: "set-me"
+
+  ## Default cert-manager issuer to use in ingresses
   issuer: letsencrypt-staging
+
+  ## Verify ingress certificates
+  ##
   verifyTls: true
   clusterDns: "10.43.0.10"
```
Comment (Xartos) [inline config/config/wc-config.yaml:14]: How do you decide on what variables to comment? I'm guessing you only comments the lines you feel necessary, but I'm curios if you have any process for deciding.

File: config/config/wc-config.yaml [hunk 9]
```diff
@@ -22,53 +27,95 @@ storageClasses:
   ebs:
     enabled: null # true | false
 
+  storageClass: "set-me"
```
Comment (Xartos) [inline config/config/wc-config.yaml:30]: Why is this here?

File: config/config/wc-config.yaml [hunk 10]
```diff
@@ -22,53 +27,95 @@ storageClasses:
...
   type: "s3"
   # gcs:
   #   project: "set-me"
   # s3:
   #   region: "set-me"
   #   regionAddress: "set-me"
   #   regionEndpoint: "set-me"
+
+  ##
+  ## TODO: remove all but for 'velero'.
```
Comment (Xartos) [inline config/config/wc-config.yaml:50]: Why remove those?

File: config/config/wc-config.yaml [hunk 11]
```diff
@@ -22,53 +27,95 @@ storageClasses:
...
   buckets:
     harbor: "${CK8S_ENVIRONMENT_NAME}-harbor"
     velero: "${CK8S_ENVIRONMENT_NAME}-velero"
     elasticsearch: "${CK8S_ENVIRONMENT_NAME}-es-backup"
     influxDB: "${CK8S_ENVIRONMENT_NAME}-influxdb"
     scFluentd: "${CK8S_ENVIRONMENT_NAME}-sc-logs"
 
+## User configuration.
 user:
   # This only controls if the namespaces should be created, user RBAC is always created.
```
Comment (Xartos) [inline config/config/wc-config.yaml:61]: Add two `#` here as well?

File: config/config/wc-config.yaml [hunk 12]
```diff
@@ -22,53 +27,95 @@ storageClasses:
...
   createNamespaces: true
+  ## List of user namespaces to create.
+  ##
   namespaces:
```
Comment (Xartos) [inline config/config/wc-config.yaml:65]: Should we make sure to leave a empty line above all these comments? (seems like that how you've done it for most of the others

File: config/config/wc-config.yaml [hunk 13]
```diff
@@ -22,53 +27,95 @@ storageClasses:
...
+
+  ## Alertmanager configuration.
+  ##
   alertmanager:
     enabled: false
     namespace: monitoring
     ingress:
       enabled: false
 
+
```
Comment (Xartos) [inline config/config/wc-config.yaml:81]: Maybe make sure exactly one empty line between

File: config/config/wc-config.yaml [hunk 14]
```diff
@@ -84,41 +131,85 @@ falco:
     tolerations: []
     affinity: {}
     nodeSelector: {}
+    ## if type=slack falco.alerts.slackWebhook must be set in the secrets yaml file
```
Comment (Xartos) [inline config/config/wc-config.yaml:134]: Shouldn't this be in the secrets file instead?

File: config/config/wc-config.yaml [hunk 15]
```diff
@@ -84,41 +131,85 @@ falco:
...
+  ## ref: https://kubernetes.io/docs/user-guide/compute-resources/
+  ##
   resources:
     requests:
       memory: 1Gi
       cpu: 300m
     limits:
       memory: 2Gi
-      cpu: "1"
+      cpu: "1" ## Must be a string
```
[the remark this answers is not in the document]
Comment (lentzi90) [reply config/config/wc-config.yaml:161]: The API-server complains. I'm not sure exactly why though, haven't looked into it, but I'm guessing that this is set on the Prometheus object and that the CRD requires it to be a string for some reason.

[the remark this answers is not in the document]
Comment (lentzi90) [reply config/config/wc-config.yaml:161]: Actually, it looks like string or integer both should be fine, but no floats though. Maybe that is just in newer versions or something?
https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/crds/crd-prometheuses.yaml#L1987

File: config/config/wc-config.yaml (hunk 10 above)
Comment (OlleLarsson) [reply config/config/wc-config.yaml:50]: Because only the velero bucket is used in the workload cluster.

File: config/config/wc-config.yaml (hunk 9 above)
Comment (OlleLarsson) [reply config/config/wc-config.yaml:30]: rebase :)

File: config/config/wc-config.yaml (hunk 15 above)
[the remark this answers is not in the document]
Comment (Xartos) [reply config/config/wc-config.yaml:161]: I just looked at it. And it seems like the CRD accepts both int and string in the new version (0.45) but I think we use 0.44 which only accepts a string. So that must be it.

File: config/config/wc-config.yaml (hunk 10 above)
Comment (Xartos) [reply config/config/wc-config.yaml:50]: Ofcorse, I didn't think about that.

File: config/config/wc-config.yaml (hunk 2 above)
Comment (viktor-f) [reply config/config/wc-config.yaml:363]: I misunderstood the comment. I thought it meant that I should set `useHostPort: 80` or `useHostPort: 443`. But here I see that it should be `true` or `false`: https://github.com/elastisys/compliantkubernetes-apps/blob/15038efd79296dde6e996233e88dbfc096dd6a0a/helmfile/values/ingress-nginx.yaml.gotmpl#L35

Maybe change it to this (or something similar):
```suggestion
    ## If `true`, nginx will use host ports 80 and 443
```

**Review by lentzi90 — APPROVED** [review]
This is great!

File: config/config/wc-config.yaml (hunk 10 above)
Comment (viktor-f) [reply config/config/wc-config.yaml:94]: Could you remove these now instead of just adding a TODO? If they are not used anywhere I don't think it should be a problem.

File: config/config/wc-config.yaml [hunk 16]
```diff
@@ -1,78 +1,158 @@
...
+##
+## tolerations: []
+
+## Affinity for pod assignment
+## ref: https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity
+##
+## affinity: {}
+
+## Some common options used in various helm charts.
+##
```
Comment (viktor-f) [inline config/config/wc-config.yaml:24]: You seem to have removed the lines with just `##` in the rest of the file, so you could do the same here.
```suggestion
## Define resources requests and limits for single Pods.
## ref: https://kubernetes.io/docs/user-guide/compute-resources/
## resources: {}

## Node labels for pod assignment
## ref: https://kubernetes.io/docs/user-guide/node-selection/
## nodeSelector: {}

## Tolerations for pod assignment
## ref: https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/
## tolerations: []

## Affinity for pod assignment
## ref: https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity
## affinity: {}

## Some common options used in various helm charts.
```

File: config/config/wc-config.yaml [hunk 17]
```diff
@@ -178,12 +304,19 @@ ck8sdash:
...
+  ## Source IP range to allow.
   whitelistRange:
     global: "0.0.0.0/0"
     ck8sdash: false
 
+## Nfs-client-provisioner configuration.
+## Deployment is controlled via 'storageClasses.nfs.enabled'.
 nfsProvisioner:
   server: ""
   path: /nfs
```
Comment (viktor-f) [inline config/config/wc-config.yaml:322]: Perhaps move this block so that it's directly below the storage config.

**Review by viktor-f — COMMENTED** [review]
Looks very nice! Just some minor things

File: config/config/wc-config.yaml (hunk 10 above)
Comment (OlleLarsson) [reply config/config/wc-config.yaml]: Ok

**Review by viktor-f — APPROVED** [review]
LGTM!

**Review by Xartos — COMMENTED** [review]
Tiny remarks, otherwise it looks great!
